src/Raven.Client/Documents/Linq/LinqPathProvider.cs

```diff
@@ -248,6 +248,22 @@ public static bool GetValueFromExpressionWithoutConversion(Expression expression
                     value = ((LambdaExpression)expression).Compile().DynamicInvoke();
                     return true;
                 case ExpressionType.Call:
+                    if (expression is MethodCallExpression mce)
+                    {
+                        var args = new Object[mce.Arguments.Count];
+                        for (var index = 0; index < mce.Arguments.Count; index++)
+                        {
+                            if (GetValueFromExpressionWithoutConversion(mce.Arguments[index], out value) == false)
+                                return false;
+                            args[index] = value;
+                        }
+                        if (mce.Method.DeclaringType == typeof(RavenQuery) &&
+                            mce.Method.Name == nameof(RavenQuery.CmpXchg))
+                        {
+                            value = new MethodCall("cmpxchg", args);
```

Comment:
Enum would be better here?

Reply:
Think about DocumentQuery here. Maybe what you actually want to create here is the type that will be used there.